### PR TITLE
Framework demonstrator

### PR DESCRIPTION
**Pull request by ktf**

### CODE REVIEW

File: Framework/Core/include/Framework/LocalRootFileService.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#include <memory>
+
+namespace o2 {
+namespace framework {
+
+/// A simple service to create ROOT files in the local folder
+class LocalRootFileService : public RootFileService {
+public:
+  std::shared_ptr<TFile> open(const char *fmt, ...) final;
+  char *format(const char *fmt, ...) final;
```
Comment (matthiasrichter) [inline Framework/Core/include/Framework/LocalRootFileService.h]: here I get the following warning
```
In file included from /home/richter/src/alisw/sw/SOURCES/O2/dev/0/Framework/Core/src/runDataProcessing.cxx:24:0:
/home/richter/src/alisw/sw/SOURCES/O2/dev/0/Framework/Core/include/Framework/LocalRootFileService.h:28:9: warning: invalid covariant return type for ‘virtual char* o2::framework::LocalRootFileService::format(const char*, ...)’
   char *format(const char *fmt, ...) final;
         ^~~~~~
In file included from /home/richter/src/alisw/sw/SOURCES/O2/dev/0/Framework/Core/include/Framework/LocalRootFileService.h:13:0,
                 from /home/richter/src/alisw/sw/SOURCES/O2/dev/0/Framework/Core/src/runDataProcessing.cxx:24:
/home/richter/src/alisw/sw/SOURCES/O2/dev/0/Framework/Core/include/Framework/RootFileService.h:32:23: note:   overriding ‘virtual const char* o2::framework::RootFileService::format(const char*, ...)’
   virtual const char *format(const char *fmt, ...) = 0;
                       ^~~~~~
```

File: Framework/Core/include/Framework/ParamRetriever.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+//
+// See https://alice-o2.web.cern.ch/ for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+#ifndef FRAMEWORK_PARAMETERRETRIEVER_H
+#define FRAMEWORK_PARAMETERRETRIEVER_H
+
+#include <string>
```
Comment (matthiasrichter) [inline Framework/Core/include/Framework/ParamRetriever.h]: need also `#include <vector>`

Comment (ktf) [reply Framework/Core/include/Framework/ParamRetriever.h]: yeah, not sure why it works on my mac.. Fixed.

File: Framework/Core/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+# 
+# See https://alice-o2.web.cern.ch/ for full licensing information.
+# 
+# In applying this license CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+set(MODULE_NAME "Framework")
+
+set(MODULE_BUCKET_NAME O2DeviceApplication_bucket)
```
Comment (matthiasrichter) [inline Framework/Core/CMakeLists.txt]: Probably that should be a separate `FrameworkCore_bucket`. The linker error comes from the missing `libNet` from ROOT where the TMessage is implemented.

Comment (ktf) [reply Framework/Core/CMakeLists.txt]: Yes, agreed. I will fix it.

**Comment by dberzano** [comment]
(Re-closing it for testing.)

**Comment by dberzano** [comment]
With alisw/alidist#837 the copyright checker should succeed.

File: Framework/Core/include/Framework/LocalRootFileService.h (hunk 1 above)
Comment (matthiasrichter) [reply Framework/Core/include/Framework/LocalRootFileService.h]: The base class `RootFileService` defines the return type of `format` as `const char*`. That is where the warning comes from.

**Comment by ktf** [comment]
Last two errors are actually a regression in FairRoot. FairRootGroup/FairRoot#627 should fix them.

**Comment by ktf** [comment]
Agreed. It's in any case self contained in Framework, so no one should be affected. I will squash all the commits, update the README with the latest version of the document and then do the merge. We can continue from there, writing user stories of what we want to achieve next in terms of features. For example porting the Test Timeframe stuff will require better support to express data parallel / time parallel pipelines.

**Pull request merged**